app/src/main/java/de/danoeh/antennapod/dialog/FeedRefreshIntervalDialog.java

```diff
@@ -0,0 +1,161 @@
+        spinnerArrayAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
+
+        int hourOfDay = 7;
+        int minute = 0;
+        int[] updateTime = UserPreferences.getUpdateTimeOfDay();
+        if (updateTime.length == 2) {
+            hourOfDay = updateTime[0];
+            minute = updateTime[1];
+        }
+
+        int finalCheckedItem = checkedItem;
+        int finalHourOfDay = hourOfDay;
+        int finalMinute = minute;
+
+        Spinner spinner = (Spinner) view.findViewById(R.id.spinner);
+        RadioButton radioButtonInterval = (RadioButton) view.findViewById(R.id.intervalRadioButton);
+        RadioButton radioButtonTime = (RadioButton) view.findViewById(R.id.timeRadioButton);
+        RadioButton radioButtonDisable = (RadioButton) view.findViewById(R.id.disableButton);
+        RadioGroup radioGroupOne = (RadioGroup) view.findViewById(R.id.genderRadioGroup);
+        TimePicker timePicker = (TimePicker) view.findViewById(R.id.timePicker);
+        radioGroupOne.setOnCheckedChangeListener(new RadioGroup.OnCheckedChangeListener() {
+            @Override
+            public void onCheckedChanged(RadioGroup radioGroup, int i) {
+                switch (i) {
+                    case R.id.intervalRadioButton:
+
+                        spinner.setAdapter(spinnerArrayAdapter);
+                        spinner.setSelection(finalCheckedItem);
+                        spinner.setVisibility(View.VISIBLE);
+                        timePicker.setVisibility(View.GONE);
+                        break;
+                    case R.id.timeRadioButton:
+                        spinner.setVisibility(View.GONE);
+                        timePicker.setVisibility(View.VISIBLE);
+
+                        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+                            timePicker.setHour(finalHourOfDay);
+                        }
+                        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+                            timePicker.setMinute(finalMinute);
+                        }
+                        timePicker.setIs24HourView(DateFormat.is24HourFormat(context));
+                        break;
+                    case R.id.disableButton:
+                        spinner.setVisibility(View.GONE);
+                        timePicker.setVisibility(View.GONE);
+                        break;
+                    default:
+
+                }
+            }
+        });
+
+        radioButtonInterval.setChecked(true);
+
+        builder.setPositiveButton(R.string.confirm_label, (dialog, which) -> {
+            if (radioButtonInterval.isChecked()) {
+                String text = spinner.getSelectedItem().toString();
+                String[] parts = text.split(" ");
+                int hours = Integer.parseInt(parts[0]);
+                UserPreferences.setUpdateInterval(hours);
+                onUpdateCallback.handleUpdate();
```

Comment:
Instead of using a callback, the settings screen could register a `OnSharedPreferenceChangeListener`. That way, there is less coupling between the classes.